apps/src/applab/AppLabView.jsx

```diff
@@ -34,8 +35,101 @@ var AppLabView = React.createClass({
     onMount: React.PropTypes.func.isRequired
   },
 
+  getInitialState() {
+    // only used so that we can rerender when resized
+    return {
+      width: undefined,
+      height: undefined
+    };
+  },
+
+  /**
+   * Called when the window resizes. Look to see if width/height changed, then
+   * call adjustHeights as our maxHeight may need adjusting.
+   */
+  onResize: function () {
+    var windowWidth = $(window).width();
+    var windowHeight = $(window).height();
+
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.width && windowHeight === this.state.height) {
+      return;
+    }
+
+    this.setState({
+      width: $(window).width(),
+      height: $(window).height()
+    });
+
+    this.adjustHeights();
+  },
+
+  /**
+   * Adjust the height/maxHeight of our instructions based on the rendered size
+   * of the instructions, and the rendered size of the workspace.
+   */
+  adjustHeights: function () {
+    if (!this.props.instructionsInTopPane) {
+      return;
+    }
+
+    var instructionsContent = this.refs.topInstructions.refs.instructions.refs.instructionsMarkdown;
+    var instructionsContentHeight = $(ReactDOM.findDOMNode(instructionsContent)).outerHeight(true);
+    var workspaceHeight = $(ReactDOM.findDOMNode(this.refs.codeWorkspace)).height();
+    var totalHeight = workspaceHeight + this.props.instructionsHeight;
+
+    var height = this.props.instructionsHeight || 300;
```

Comment:
This probably deserves a comment as well as a constant. 300 is the default height, but we don't want to set it before our initial render (i.e. don't want to just have it as the initialState in the reducer). I forget exactly why, but that's further indication that this needs an explanation.